pkg/kube/inject/inject.go

```diff
@@ -1052,3 +1056,41 @@ func rewriteCniPodSpec(annotations map[string]string, spec *SidecarInjectionSpec
 		}
 	}
 }
+
+// overwriteClusterInfo updates cluster name and network from url path
+// This is needed when webconfig config runs on a different cluster than webhook
+func overwriteClusterInfo(containers []corev1.Container, path string) {
+	if strings.HasPrefix(path, "/inject/") {
+		// split path
+		// TODO dynamic read network from cluster name
+		log.Infof("overwriting cluster info based on path: %s\n", path)
+
+		path = strings.TrimPrefix(path, "/inject/")
+		res := strings.Split(path, "/")
+		clusterName, clusterNetwork := "", ""
+		if len(res) == 2 {
+			clusterName = res[0]
+			clusterNetwork = res[1]
+		} else if len(res) == 1 {
+			clusterName = res[0]
+		}
+
+		log.Debugf("overwriting cluster info based on clusterName: %s clusterNetwork: %s\n", clusterName, clusterNetwork)
+
+		for _, c := range containers {
+			if c.Name == ProxyContainerName {
+				for _, e := range c.Env {
+					if e.Name == "ISTIO_META_CLUSTER_ID" && clusterName != "" {
```

Comment:
Not sure this is what we want - if the user explicitly set the META in the pod - you should let it be.

If it is not set in the pod ( common case ) - set it with the values from the URL. 

Pretty sure the pods don't have this by default.

Reply:
You are right, by default this is set to Kubernetes from reading the injection ConfigMap.  I've reworked the code to behave correctly, where the code would overwrite it with the cluster name and network from webhook url path.